.changeset/polite-tips-attend/changes.md

```diff
@@ -0,0 +1,15 @@
+Added createStyled overload to handle when shouldForwardProp is a custom type guard for intrinsic props
```

Comment:
```suggestion
Added `CreateStyled` overload to handle when `shouldForwardProp` is a custom type guard for intrinsic props.
```

Reply:
I've wanted to commit this suggestion through GitHub interface - but seems like I don't have sufficient permissions to push to your fork.

You can check how to allow this for the future PRs here - https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork . But at the moment just commit this suggestion with a click on a button.